resources-manager/src/AdaptivePruningManager.cpp

```diff
@@ -0,0 +1,81 @@
+#include "AdaptivePruningManager.hpp"
+#include "Bitmap.hpp"
+#include "ReplicaConfig.hpp"
+#include "concord.cmf.hpp"
+#include "SigManager.hpp"
+
+using namespace concord::performance;
+
+AdaptivePruningManager::AdaptivePruningManager(
+    const std::shared_ptr<concord::performance::IResourceManager> &resourceManager,
+    const std::chrono::duration<double, std::milli> &interval,
+    const std::shared_ptr<bftEngine::impl::IInternalBFTClient> &bftClient)
+    : repId(bftEngine::ReplicaConfig::instance().getreplicaId()),
+      resourceManager(resourceManager),
+      mode(LEGACY),
+      interval(interval),
+      bftClient(bftClient) {
+  start();
+}
+
+AdaptivePruningManager::~AdaptivePruningManager() { stop(); }
+
+void AdaptivePruningManager::notifyReplicas(const long double &rate, const uint64_t batchSize) {
+  concord::messages::ReconfigurationRequest rreq;
+
+  concord::messages::PruneTicksChangeRequest pruneRequest;
+  pruneRequest.sender_id = bftClient->getClientId();
+  pruneRequest.tick_period_seconds =
+      rate / bftClient->numOfConnectedReplicas(bftEngine::ReplicaConfig::instance().numReplicas);
+  pruneRequest.batch_blocks_num = batchSize;
```

Comment:
if the resolution is in seconds, it means that at most a replica can generate 1 tick per second? 
I think that because the resource manager doesn't know about the ticks resolution and cluster size.
it's better for it to return only blocks per second, and the manager should convert it to ticks resolution and batch size. 
for example, if the resource manager calculates 20 blocks per second, it can't calculate the batch size as it is dependent on the cluster size and tick resolution.

Reply:
As below 1 tick per second is the minimum not maximum. About, blocks per second - it was this way, but it was agreed that we change it to PruneInfo (ticks per second, batchSzie). I can change it back to what it was, but we are running out of time.